badook_tests/dsl/summary.py

```diff
@@ -20,10 +21,19 @@ def __init__(self, ctx):
         self.summaries = {}
         self.joined_datasets = {}
         self.user_features = {}
+        self._featuresets_def = {}
```

Comment:
```suggestion
        self._feature_sets_def = {}
```
Also, please add type annotations.